pkg/apiserver/storage/ram/store.go

```diff
@@ -224,23 +227,35 @@ func (s *store) Watch(ctx context.Context, key string, labelSelector labels.Sele
 	s.eventMutex.RLock()
 	defer s.eventMutex.RUnlock()
 
+	selectors := &antreastorage.Selectors{
+		Key:   key,
+		Label: labelSelector,
+		Field: fieldSelector,
+	}
+
 	allObjects := s.storage.List()
-	initEvents := make([]antreastorage.InternalEvent, len(allObjects))
-	for i, obj := range allObjects {
+	initEvents := make([]antreastorage.InternalEvent, 0, len(allObjects))
+	for _, obj := range allObjects {
 		// Objects retrieved from storage have been verified with keyFunc when they are inserted.
 		key, _ := s.keyFunc(obj)
+		// Check whether the watcher is interested in this object, don't generate it as an initEvent
+		// if not.
+		if s.selectFunc != nil && !s.selectFunc(selectors, key, obj) {
```

Comment:
Question - if we filter here, do we still need to filter in ToWatchEvent()?

Reply:
Good catch! The filter will be executed for init events twice at the moment, I didn't find good way to let ToWatchEvent know the information yet, will try to avoid it today.